server/src/main/java/io/crate/statistics/TableStats.java

```diff
@@ -62,6 +65,18 @@ public long estimatedSizePerRow(RelationName relationName) {
         return tableStats.getOrDefault(relationName, Stats.EMPTY).averageSizePerRowInBytes();
     }
 
+    public long estimatedSizePerRow(TableInfo tableInfo) {
+        Stats stats = tableStats.get(tableInfo.ident());
+        if (stats == null) {
+            // if stats are not available we fall back to estimate the size based on
+            // column types. Therefore we need to get the column information.
+            Collection<Reference> ramAccountedColumns = tableInfo.allColumns();
+            return Stats.EMPTY.estimateSizeForColumns(ramAccountedColumns);
```

Comment:
This will need to account for the changes in https://github.com/crate/crate/pull/17747